src/widgets/analytics/analytics.ts

```diff
@@ -0,0 +1,202 @@
+        const values = parameters[parameter].join('+');
+
+        refinements.push(
+          `${encodeURIComponent(parameter)}=${encodeURIComponent(
+            parameter
+          )}_${encodeURIComponent(values)}`
+        );
+      }
+    }
+
+    return refinements.join('&');
+  };
+
+  const serializeNumericRefinements = function(
+    numericRefinements: any
+  ): string {
+    const refinements: string[] = [];
+
+    for (const attribute in numericRefinements) {
+      if (numericRefinements.hasOwnProperty(attribute)) {
+        const filter = numericRefinements[attribute];
+
+        if (filter.hasOwnProperty('>=') && filter.hasOwnProperty('<=')) {
+          if (filter['>='][0] === filter['<='][0]) {
+            refinements.push(`${attribute}=${attribute}_${filter['>=']}`);
+          } else {
+            refinements.push(
+              `${attribute}=${attribute}_${filter['>=']}to${filter['<=']}`
+            );
+          }
+        } else if (filter.hasOwnProperty('>=')) {
+          refinements.push(`${attribute}=${attribute}_from${filter['>=']}`);
+        } else if (filter.hasOwnProperty('<=')) {
+          refinements.push(`${attribute}=${attribute}_to${filter['<=']}`);
+        } else if (filter.hasOwnProperty('=')) {
+          const equals: string[] = [];
+
+          for (const equal in filter['=']) {
+            // eslint-disable-next-line max-depth
+            if (filter['='].hasOwnProperty(equal)) {
+              equals.push(filter['='][equal]);
+            }
+          }
+
+          refinements.push(`${attribute}=${attribute}_${equals.join('-')}`);
+        }
+      }
+    }
+
+    return refinements.join('&');
+  };
+
+  let lastSentData = '';
+
+  const sendAnalytics = function(analyticsState: AnalyticsState | null): void {
+    if (analyticsState === null) {
+      return;
+    }
+
+    const serializedParams: string[] = [];
+
+    const serializedRefinements = serializeRefinements({
+      ...analyticsState.state.disjunctiveFacetsRefinements,
+      ...analyticsState.state.facetsRefinements,
+      ...analyticsState.state.hierarchicalFacetsRefinements,
+    });
+
+    const serializedNumericRefinements = serializeNumericRefinements(
+      analyticsState.state.numericRefinements
+    );
+
+    if (serializedRefinements !== '') {
+      serializedParams.push(serializedRefinements);
+    }
+
+    if (serializedNumericRefinements !== '') {
+      serializedParams.push(serializedNumericRefinements);
+    }
+
+    const stringifiedParams = serializedParams.join('&');
+
+    let dataToSend = `Query: ${
+      analyticsState.state.query
+    }, ${stringifiedParams}`;
+    if (pushPagination === true) {
+      dataToSend += `, Page: ${analyticsState.state.page}`;
+    }
+
+    if (lastSentData !== dataToSend) {
+      pushFunction(
+        stringifiedParams,
+        analyticsState.state,
+        analyticsState.results
+      );
+
+      lastSentData = dataToSend;
+    }
+  };
+
+  let pushTimeout: any;
```

Comment:
To be more specific, it could be typed as `number` and use `window.setTimeout` (to be coherent with `history.ts`)